libzkbob-rs/src/merkle.rs

```diff
@@ -694,23 +885,148 @@ impl<D: KeyValueDB, P: PoolParams> MerkleTree<D, P> {
                 index /= 2;
             }
             if nodes_request_index < clean_index {
```

Comment:
> it seems that clean_index is only used in test currently, so it really always equals 0, and therefore comparison doesn't really makes sense

Yes, `clean_index` is the old Zeropool workaround to reduce the tree size. It's supposed that some set of nodes can be removed with restore ability. Currently it isn't in use. I decide do not remove it now to prevent additional codebase changes. But it make sense to remove that functional during the next code refactoring (which is ripe already)